### PR TITLE
added move contribution feature

### LINKED ISSUE #313: Ability to move contributions
We want to enable an option to super admin so that whenever super admin hovers on a contribution by holding the command/control key we should show a settings icon.

Upon clicking on the settings icon, we should be able to do the following

- open up a modal that allows the super admin to either move that contribution to noteworthy contribution or archive that contribution or move to `Other contributions` (we need to create this type)

[Ask Sumit for more details]

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
### Working on a PR to have a context that will handle the key press for all  the page

- https://github.com/Real-Dev-Squad/website-members/issues/420

### What is the change?
closes #313 . It allows the super user to move contributions from noteworthy to other, and from other to noteworthy

### Is it bug? 
no

### \*Dev Tested?

- [X] Yes
- [ ] No

#### Platforms

- [X] Web

#### Browsers

- [X] Chrome
- [ ] Safari
- [ ] Firefox

### Before / After Change Screenshots

> For visual or interaction changes. Can be video / screenshot.

https://user-images.githubusercontent.com/90315175/184618250-6d2157ad-4fc7-4868-921d-3b4962ab7ab3.mp4

### CODE REVIEW

File: src/components/member-profile/contribution/index.js [hunk 1]
```diff
@@ -63,11 +66,19 @@ const ContributionCard = ({
...
-    task: { title, startedOn, endsOn, status, purpose },
+    task: { id, title, startedOn, endsOn, status, purpose, isNoteworthy },
     prList,
   } = contribution;
   const isTitleAvailable = !!title;
   const featureTitle = isTitleAvailable ? title : prList[0].title;
+  const [showSettings, setShowSettings] = useState(false);
+  const { isOptionKey, setIsOptionKey } = userContext();
+
+  const handleSettingsButton = () => {
```
Comment (samarpan1738) [inline src/components/member-profile/contribution/index.js:77]: Nit: Method name can be better
```suggestion
  const revealSettingsButton = () => {
```

File: src/components/member-profile/contribution/index.js [hunk 2]
```diff
@@ -119,7 +130,13 @@ const ContributionCard = ({
     }
   }
   return (
-    <div>
+    <div
+      onMouseEnter={handleSettingsButton}
+      onMouseLeave={() => {
+        setShowSettings(false);
+        setIsOptionKey(false);
+      }}
```
Comment (samarpan1738) [inline src/components/member-profile/contribution/index.js:138]: Can we also extract this in a method?

File: src/components/member-profile/index.js [hunk 3]
```diff
@@ -351,12 +359,22 @@ const Profile = (props) => {
...
 
   return (
-    <>
+    <div
+      role="button"
+      tabIndex={0}
+      onKeyDown={(e) => {
+        if (e.ctrlKey || e.metaKey) {
+          setIsOptionKey(true);
+        }
```
Comment (samarpan1738) [inline src/components/member-profile/index.js:368]: In [HomePage](https://github.dev/vinayak-trivedi/website-members/blob/a489d33d2264767a74c8e5841f1f25bb46c33c18/src/components/home/index.js#L29) we are checking for the `Alt` key to call `setIsOptionKey(true)` and here we're using `ctrlKey` or `metaKey`. 
Should we keep it same at both places or it was intended to be different?
If we can keep it same then we should also extract this to a util method

File: src/components/member-profile/contribution/index.js (hunk 2 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:138]: yes I will do that

File: src/components/member-profile/index.js (hunk 3 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:368]: It was mentioned in the issue so I did that, but you are right it would be better if we have the same key

File: src/components/member-profile/contribution/index.js [hunk 4]
```diff
@@ -135,6 +152,13 @@ const ContributionCard = ({
           )}
         </div>
         <div className={classNames.rightSection}>
+          {isNoteworthy !== undefined ? (
```
Comment (rohan09-raj) [inline src/components/member-profile/contribution/index.js:155]: You can use `&&` instead of having if-else if you have only condition in which you want to conditionally render components or elements

File: src/components/member-profile/contribution/index.js [hunk 5]
```diff
@@ -135,6 +152,13 @@ const ContributionCard = ({
...
+            <SuperUserOptions
+              showSettings={showSettings}
+              isNoteworthy={isNoteworthy}
+              TaskId={id}
```
Comment (rohan09-raj) [inline src/components/member-profile/contribution/index.js:159]: Can it be made camelcase ??

File: src/components/member-profile/index.js [hunk 6]
```diff
@@ -351,12 +359,22 @@ const Profile = (props) => {
...
+      }}
+    >
+      {dev && <UserProfile />}
       {showModal && (
         <Modal style={modalStyle} show={showModal} closeModal={closeModal}>
           {children}
         </Modal>
       )}
+      <div>{showMemberTaskUpdateModal && <MemberTaskUpdate />}</div>
```
Comment (rohan09-raj) [inline src/components/member-profile/index.js:377]: Is the `div` required here ?

File: src/store/user/user-context.js [hunk 7]
```diff
@@ -10,17 +10,30 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [showMemberTaskUpdateModal, setshowMemberTaskUpdateModal] =
+    useState(false);
```
Comment (rohan09-raj) [inline src/store/user/user-context.js:14]: Can we have this in the same line i.e. 13 please ?

File: src/store/user/user-context.js [hunk 8]
```diff
@@ -10,17 +10,30 @@ export const UserContextProvider = ({ children }) => {
...
+  const [isnoteworthy, setIsnoteWorthy] = useState(false);
```
Comment (rohan09-raj) [inline src/store/user/user-context.js:15]: Can we please follow CamelCase here?

File: src/helper-functions/action-handlers.js [hunk 9]
```diff
@@ -20,4 +21,16 @@ const archiveMember = (user) =>
     withCredentials: true,
   });
 
-export { archiveMember, moveToMember, getUserSelf };
+const moveTask = (taskId, data) =>
+  fetch(
+    getTaskUpdateURL(taskId),
+    'patch',
```
Comment (rohan09-raj) [inline src/helper-functions/action-handlers.js:27]: Can you please check if there is any constant for it ? If not, can we please make one ?

File: src/components/member-task-update/member-task-update.module.scss [hunk 10]
```diff
@@ -0,0 +1,15 @@
+.changeTaskType {
+  margin: 10px;
+  padding: 10px;
+  font-size: inherit;
+  cursor: pointer;
+  font-weight: 600;
+  border: 1px solid grey;
```
Comment (rohan09-raj) [inline src/components/member-task-update/member-task-update.module.scss:7]: Can we please make a HEX constant for this and use it here ?

File: src/components/member-task-update/member-task-update.module.scss [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  margin: 10px;
+  padding: 10px;
+  font-size: inherit;
+  cursor: pointer;
+  font-weight: 600;
+  border: 1px solid grey;
+}
+
+.updateText {
+  color: grey;
```
Comment (rohan09-raj) [inline src/components/member-task-update/member-task-update.module.scss:11]: Here too, use a SCSS variable

File: src/components/member-task-update/index.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    const data = { isNoteworthy: !isnoteworthy };
+    const res = await moveTask(taskid, data);
+    setIsUpdating(false);
+    if (res.status === 204) {
+      setUpdateStatus(
+        `Task changed to ${
+          isnoteworthy ? 'other contribution' : 'noteworthy'
+        }! reloading...`
+      );
+      window.location.reload();
```
Comment (rohan09-raj) [inline src/components/member-task-update/index.js:30]: Is hard reload required here ? Is there any alternative to do it ?

File: src/components/member-profile/super-user-options.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+          <img
+            className={classNames.settingsIcon}
+            src="/icons/settings.png"
+            alt="setting"
+          />
+        </button>
+      </div>
+    );
+  };
+  return <>{isSuperUserMode && showSuperUserOptions()}</>;
```
Comment (samarpan1738) [inline src/components/member-profile/super-user-options.js:42]:
```suggestion
  if(isSuperUserMode)
    return (
      <div
        className={`${classNames.settingsContainer} ${
          !showSettings && classNames.hidden
        }`}
      >
        <button
          className={classNames.settingsButton}
          type="button"
          onClick={showModal}
        >
          <img
            className={classNames.settingsIcon}
            src="/icons/settings.png"
            alt="setting"
          />
        </button>
      </div>
    );
  };
  return <></>;
```

File: src/components/member-profile/super-user-options.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    isSuperUserMode,
+    showMemberTaskUpdateModal,
+    setshowMemberTaskUpdateModal,
+    setTaskId,
+    setIsnoteWorthy,
+  } = userContext();
+
+  const showModal = (e) => {
+    e.stopPropagation();
+    setshowMemberTaskUpdateModal(!showMemberTaskUpdateModal);
```
Comment (samarpan1738) [inline src/components/member-profile/super-user-options.js:16]: Shouldn't we pass `true` . Currently this is more like a `toggleModal` than `showModal`
```suggestion
    setshowMemberTaskUpdateModal(true);
```

File: src/components/member-profile/contribution/index.js (hunk 4 above)
Comment (samarpan1738) [inline src/components/member-profile/contribution/index.js:155]: Can we move this check inside `SuperUserOptions` component? We are already passing `isNoteworthy` prop

File: src/store/user/user-context.js [hunk 15]
```diff
@@ -10,17 +10,30 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [showMemberTaskUpdateModal, setshowMemberTaskUpdateModal] =
```
Comment (heyrandhir) [inline src/store/user/user-context.js:13]: "setshowMemberTaskUpdateModal" can be camelCased.

File: src/components/member-profile/index.js (hunk 3 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:368]: I checked it in the home component there is a lot of prop drilling that is happening and in my case I am taking it from usercontext, I will change that button to Alt but the refactor can be done in a future PR.

File: src/components/member-profile/contribution/index.js (hunk 4 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:155]: I want to do this because I want to render that component when noteworthy is not undefined which will be the case if the contribution is  a PR and I don't want to show it for that, other than that even if it's true or false I want to render that component

File: src/components/member-profile/contribution/index.js (hunk 5 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:159]: sure

File: src/components/member-profile/index.js (hunk 6 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:377]: yes it can be removed

File: src/store/user/user-context.js (hunk 7 above)
Comment (vinayak-trivedi) [reply src/store/user/user-context.js:14]: will do that

File: src/components/member-profile/super-user-options.js (hunk 14 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/super-user-options.js:16]: yes correct, thanks for pointing out

File: src/components/member-profile/super-user-options.js (hunk 13 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/super-user-options.js:42]: will do that

File: src/components/member-profile/index.js (hunk 3 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:368]: Hey samarpan I tried using the Alt key but it is creating a weird behaviour in chrome, when I press the alt key it shifts the focus to customize and control google chrome features, so for the first time it works but if the user tries to access the buttons again with pressing that alt key the setting icon doesn't show up. 
Image for reference⬇️
image

File: src/components/member-profile/contribution/index.js (hunk 1 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:77]: will do that

File: src/store/user/user-context.js (hunk 8 above)
Comment (vinayak-trivedi) [reply src/store/user/user-context.js:15]: will do that

File: src/components/member-task-update/member-task-update.module.scss (hunk 11 above)
Comment (vinayak-trivedi) [reply src/components/member-task-update/member-task-update.module.scss:11]: I will do that

File: src/components/member-profile/contribution/index.js (hunk 4 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:155]: I want to check if it is undefined and based on that I want to render it, because it should show up only for the tasks and not for the PR's

File: src/store/user/user-context.js (hunk 15 above)
Comment (vinayak-trivedi) [reply src/store/user/user-context.js:13]: will do that

File: src/store/user/user-context.js (hunk 7 above)
Comment (vinayak-trivedi) [reply src/store/user/user-context.js:14]: @rohan09-raj it is a problem with prettier, when I run yarn lint it changes it, and if I change it again it is not allowing me to commit

**Comment by heyrandhir** [comment]
LGTM 👍🏻

File: src/components/member-profile/contribution/index.js [hunk 16]
```diff
@@ -63,11 +66,24 @@ const ContributionCard = ({
...
+  const revealSettingButton = () => {
+    if (isOptionKey) {
+      setShowSettings(true);
+    }
+  };
+
+  const hideSettingButton = () => {
+    setShowSettings(false);
+    setIsOptionKey(false);
+  };
```
Comment (ankushdharkar) [inline src/components/member-profile/contribution/index.js:86]: We will be using the 'option key' quite often. Instead of every component having to write their own handlers with the same kind of code, I am wondering if we could write a reusable logic that can be [_injected_](https://en.wikipedia.org/wiki/Dependency_injection) into the component?

Randhir did something similar in [this PR for reference](https://github.com/Real-Dev-Squad/website-members/pull/410).

File: src/components/member-profile/index.js [hunk 17]
```diff
@@ -11,6 +11,10 @@ import ShowSkills from '@components/member-card/show-skills';
 import { useForm } from 'react-hook-form';
 import { isEmail, isDecimal } from 'validator';
 import { KEY_ESC, KEY_TAB } from '@constants/AppConstants';
+import MemberTaskUpdate from '@components/member-task-update';
+import { userContext } from '@store/user/user-context';
```
Comment (ankushdharkar) [inline src/components/member-profile/index.js:15]: Maybe not relevant to your PR, but a question for future: Does the file need to be called `user-context`? It's already in the `user` folder.

File: src/store/user/user-context.js [hunk 18]
```diff
@@ -10,17 +10,30 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [showMemberTaskUpdateModal, setShowMemberTaskUpdateModal] =
+    useState(false);
+  const [isNoteworthy, setIsNoteworthy] = useState(false);
+  const [isOptionKey, setIsOptionKey] = useState(false);
+  const [taskId, setTaskId] = useState('');
```
Comment (ankushdharkar) [inline src/store/user/user-context.js:17]: Should this belong in the `user-context`?

File: src/components/member-task-update/index.js [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+    } else {
+      setUpdateStatus(`There was an error while changing the task`);
+    }
+  };
+
+  const renderTaskUpdateButtton = () => {
+    const task = isNoteworthy ? 'Other contribution' : 'noteworthy';
```
Comment (ankushdharkar) [inline src/components/member-task-update/index.js:37]: These harded values should come from an enum in the constants file.

File: src/components/member-task-update/index.js [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+
+  const changeTaskType = async (taskid) => {
+    setIsUpdating(true);
+    const data = { isNoteworthy: !isNoteworthy };
+    const res = await moveTask(taskid, data);
+    setIsUpdating(false);
+    if (res.status === 204) {
+      setUpdateStatus(
+        `Task changed to ${
+          isNoteworthy ? 'other contribution' : 'noteworthy'
```
Comment (ankushdharkar) [inline src/components/member-task-update/index.js:27]: These harded values should come from an enum in the constants file.

File: src/components/member-profile/index.js [hunk 21]
```diff
@@ -67,6 +71,10 @@ const Profile = (props) => {
     'linkedin_id',
     'instagram_id',
   ];
+  const { showMemberTaskUpdateModal, setIsOptionKey } = userContext();
+
+  const { query } = useRouter() || { query: { dev: false } };
```
Comment (ankushdharkar) [inline src/components/member-profile/index.js:76]: I feel we don't need to do this part ` || { query: { dev: false } }`

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: src/components/member-profile/contribution/index.js (hunk 16 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:86]: Okay I will look at it

File: src/components/member-task-update/index.js (hunk 19 above)
Comment (vinayak-trivedi) [reply src/components/member-task-update/index.js:37]: I will do that

File: src/components/member-profile/index.js (hunk 21 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:76]: okay I will remove it.

File: src/components/member-task-update/index.js (hunk 20 above)
Comment (vinayak-trivedi) [reply src/components/member-task-update/index.js:27]: I wil do that

File: src/store/user/user-context.js (hunk 18 above)
Comment (vinayak-trivedi) [reply src/store/user/user-context.js:17]: We were having the showModal context here only so I thought that it could be there but you are right we can make a different context for it

File: src/components/member-profile/index.js (hunk 17 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js:15]: The user folder also has an index file that exports member context, so I guess it is good to have the file name as user-context.

Comment (vinayak-trivedi) [reply src/components/member-profile/index.js]: That PR doesn't contain those changes, but will refactor the code in a different PR as discussed

File: src/components/member-profile/contribution/index.js (hunk 16 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js]: That PR doesn't contain those changes, but will refactor the code in a different PR as dicussed

File: src/components/member-profile/contribution/index.js [hunk 22]
```diff
@@ -63,11 +70,13 @@ const ContributionCard = ({
...
 }) => {
   const {
-    task: { title, startedOn, endsOn, status, purpose },
+    task: { id, title, startedOn, endsOn, status, purpose, isNoteworthy },
     prList,
   } = contribution;
   const isTitleAvailable = !!title;
   const featureTitle = isTitleAvailable ? title : prList[0].title;
+  const [showSettings, setShowSettings] = useState(false);
+  const { isOptionKey, setIsOptionKey } = useTaskContext();
```
Comment (vinayak-trivedi) [inline src/components/member-profile/contribution/index.js:79]: This context is going to be changed to another context and then we don't need to take care of the option key for every component separately. Issue for reference⬇️
- https://github.com/Real-Dev-Squad/website-members/issues/420

File: src/components/member-profile/index.js [hunk 23]
```diff
@@ -351,12 +359,22 @@ const Profile = (props) => {
...
   return (
-    <>
+    <div
+      role="button"
+      tabIndex={0}
+      onKeyDown={(e) => {
+        if (e.altKey) {
+          setIsOptionKey(true);
+        }
+      }}
```
Comment (vinayak-trivedi) [inline src/components/member-profile/index.js:369]: This will be removed when this issue is closed 
- https://github.com/Real-Dev-Squad/website-members/issues/420

File: src/store/tasks/tasks-context.js [hunk 24]
```diff
@@ -0,0 +1,36 @@
+import { createContext, useState, useContext } from 'react';
+
+const TaskContext = createContext();
+
+export const TaskContextProvider = ({ children }) => {
+  const [showMemberTaskUpdateModal, setShowMemberTaskUpdateModal] =
+    useState(false);
+  const [isNoteworthy, setIsNoteworthy] = useState(false);
+  const [isOptionKey, setIsOptionKey] = useState(false);
```
Comment (vinayak-trivedi) [inline src/store/tasks/tasks-context.js:9]: This will be removed after this issue is closed
- https://github.com/Real-Dev-Squad/website-members/issues/420

File: src/constants/AppConstants.js [hunk 25]
```diff
@@ -65,4 +65,8 @@ export const HOST_NAME = 'members.realdevsquad.com';
 export const TIMEOUT = 1000;
 export const KEY_TAB = 9;
 export const KEY_ESC = 27;
+export const TASK_TYPE = {
+  ISNOTEWORTHY: 'Noteworthy',
```
Comment (heyrandhir) [inline src/constants/AppConstants.js:69]: shouldn't the key be `NOTEWORTHY` instead of `ISNOTEWORTHY `

**Review by heyrandhir — COMMENTED** [review]
LGTM. Just a small change requested.

**Review by heyrandhir — APPROVED** [review]
LGTM 👍🏻

File: src/helper-functions/show-super_user-options.js [hunk 26]
```diff
@@ -0,0 +1,9 @@
+export const revealSuperUserOption = (setShowOption, isOptionKey) => {
```
Comment (shubham-y) [inline src/helper-functions/show-super_user-options.js:1]: NIT
```suggestion
export const showSuperUserOption = (setShowOption, isOptionKey) => {
```

File: src/store/tasks/tasks-context.js [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+      {children}
+    </TaskContext.Provider>
+  );
+};
+
+export const useTaskContext = () => {
+  const context = useContext(TaskContext);
+  if (!context)
+    throw new Error(`taskcontext context can only  
+        be used in a component wrapped with taskcontext`);
```
Comment (shubham-y) [inline src/store/tasks/tasks-context.js:31]:
```suggestion
    throw new Error(`taskContext context can only  
        be used in a component wrapped with taskContext`);
```

File: src/components/member-task-update/index.js [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+    showMemberTaskUpdateModal,
+    setShowMemberTaskUpdateModal,
+    isNoteworthy,
+    taskId,
+  } = useTaskContext();
+
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [updateStatus, setUpdateStatus] = useState('');
+
+  const changeTaskType = async (taskid) => {
```
Comment (shubham-y) [inline src/components/member-task-update/index.js:20]:
```suggestion
  const changeTaskType = async (taskId) => {

```

File: src/components/member-task-update/index.js [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+    setIsUpdating(false);
+    if (res.status === 204) {
+      setUpdateStatus(
+        `Task changed to ${
+          isNoteworthy ? TASK_TYPE.OTHER : TASK_TYPE.NOTEWORTHY
+        }! reloading...`
+      );
+      window.location.reload();
+    } else {
+      setUpdateStatus(`There was an error while changing the task`);
```
Comment (shubham-y) [inline src/components/member-task-update/index.js:33]: NIT
```suggestion
      setUpdateStatus(`There was an error while updating the task`);

```

File: src/components/member-task-update/index.js [hunk 30]
```diff
@@ -0,0 +1,70 @@
...
+
+    if (updateStatus === '') {
+      return (
+        <div>
+          <button
+            className={classNames.changeTaskType}
+            type="button"
+            onClick={() => changeTaskType(taskId)}
+          >
+            Change Task to {task}
```
Comment (shubham-y) [inline src/components/member-task-update/index.js:48]: NIT
```suggestion
            Move Task to {task}

```

File: src/components/member-profile/index.js [hunk 31]
```diff
@@ -351,12 +355,13 @@ const Profile = (props) => {
   };
 
   return (
-    <>
+    <div>
```
Comment (shubham-y) [inline src/components/member-profile/index.js:358]: Do we need this extra `div`?

File: src/components/member-task-update/index.js (hunk 28 above)
Comment (vinayak-trivedi) [reply src/components/member-task-update/index.js:20]: can't do this, because there is another taskId in that scope, so yarn lint gives an error

File: src/components/member-profile/index.js (hunk 31 above)
Comment (vinayak-trivedi) [reply src/components/member-profile/index.js]: removed it

**Review by shubham-y — APPROVED** [review]
LGTM 👍

File: src/components/member-profile/contribution/index.js [hunk 32]
```diff
@@ -43,7 +50,6 @@ const Contribution = ({ contribution, fullName, imageLink, devUser }) => {
       <div
         className={url && classNames.contributionCard}
         onClick={gotoUrl}
-        onKeyDown={gotoUrl}
         aria-hidden="true"
```
Comment (vinayak-trivedi) [inline src/components/member-profile/contribution/index.js:53]: removed this as it was causing issue in our super user option, because of this the link was opening up when we had alt key pressed.

File: src/components/member-profile/super-user-options/super-user-options.js [hunk 33]
```diff
@@ -0,0 +1,41 @@
...
+  const showModal = (e) => {
+    e.stopPropagation();
+    setShowMemberTaskUpdateModal(true);
+    setTaskId(taskId);
+    setIsNoteworthy(isNoteworthy);
+  };
+
+  if (isNoteworthy === undefined) {
+    return null;
+  }
```
Comment (vinayak-trivedi) [inline src/components/member-profile/super-user-options/super-user-options.js:18]: when contribution is a PR, it will not contain isNoteworthy

File: src/components/member-profile/contribution/index.js [hunk 34]
```diff
@@ -5,6 +5,8 @@ import classNames from '@components/member-profile/contribution/contribution.mod
 import PRLink from '@components/member-profile/contribution/pr-link';
 import { timeWas } from '@helper-functions/time-was';
 import { HOST_NAME } from '@constants/AppConstants';
+import { useState } from 'react';
+import SuperUserOptions from '../super-user-options/super-user-options-container';
 
 const renderPRLinks = (prList) =>
   prList.map(({ url }, index) => {
```
Comment (Abhay5855) [inline src/components/member-profile/contribution/index.js:12]: @vinayak-trivedi  We usually don't pass index as a key, you can check here -
https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66

Comment (vinayak-trivedi) [reply src/components/member-profile/contribution/index.js:12]: okay this is not a change that I've made abhay, I agree this need to be refactored, this is a concern of another PR

**Comment by rupal-2505** [comment]
LGTM

File: src/pages/_app.js [hunk 35]
```diff
@@ -9,6 +9,7 @@ import { MembersProvider } from '@store';
 import { UserContextProvider } from '@store/user/user-context';
 import { SearchMemberProvider } from '@store/search-members/searchMembers-context';
```
Comment (Maheima) [inline src/pages/_app.js:10]: can we sort the imports here somewhat like in the format where
the library imports are first
then components
then CSS followed by assets

or should it go inside another PR?

**Review by Maheima — CHANGES_REQUESTED** [review]

File: src/pages/_app.js (hunk 35 above)
Comment (vinayak-trivedi) [reply src/pages/_app.js:10]: I think this can be done in another PR

**Review by Maheima — APPROVED** [review]
